iac-extensions/spring-config/src/main/java/org/sonar/iac/springconfig/parser/yaml/SpringConfigYamlPreprocessor.java

```diff
@@ -27,23 +28,23 @@ public class SpringConfigYamlPreprocessor {
   private static final Pattern LINE_SEPARATORS = Pattern.compile("(\n|\r|\r\n|\u2028|\u2029)");
```

Comment:
I'm expecting this regex to not work as expected.
This would match `\r` before it has a chance to match `\r\n`, transforming `data1\r\ndata2` into `['data1', '\ndata2']`.
This is not related to this PR, so it's probably better to handle it in a separate ticket/PR.
```suggestion
  private static final Pattern LINE_SEPARATORS = Pattern.compile("(\r\n|\n|\r|\u2028|\u2029)");
```

Reply:
Good catch! I added a quick test case for it as well to this PR.